src/prefect/serialization/run_config.py

```diff
@@ -32,6 +32,7 @@ class Meta:
     memory_request = fields.String(allow_none=True)
     service_account_name = fields.String(allow_none=True)
     image_pull_secrets = fields.List(fields.String(), allow_none=True)
+    always_pull_new_image = fields.String(allow_none=False)
```

Comment:
This should be `fields.Boolean`, I think this is why "False == False" wasn't working in your test case